src/LinqToExcel.Tests/InvalidCastTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace LinqToExcel.Tests
+{
+    [Author("Alberto Chvaicer", "achvaicer@gmail.com")]
+    [Category("Integration")]
+    [TestFixture]
+    public class InvalidCastTests
+    {
+        private IExcelQueryFactory _factory;
+        private string _excelFileName;
+
+        [OneTimeSetUp]
+        public void fs()
+        {
+            var testDirectory = AppDomain.CurrentDomain.BaseDirectory;
+            var excelFilesDirectory = Path.Combine(testDirectory, "ExcelFiles");
+            _excelFileName = Path.Combine(excelFilesDirectory, "Companies.xls");
+        }
+
+        [SetUp]
+        public void s()
+        {
+            _factory = new ExcelQueryFactory(_excelFileName, new LogManagerFactory());
+        }
+
+        [Test]
+        public void invalid_number_cast_with_header()
+        {
+            Assert.That(() => (from x in ExcelQueryFactory.Worksheet<Company>("Invalid Cast", _excelFileName, new LogManagerFactory())
+                         select x).ToList(), Throws.TypeOf<LinqToExcel.Exceptions.ExcelException>(), "Error on row 8 and column name 'EmployeeCount'.");
+        }
+
+        [Test, Ignore("Cast is not working")]
```

Comment:
What do you mean it's not working? Can you clarify?

Reply:
The query with Cast<double>() is not returning double values.
If it returned, would throw an exception because one of the values is the string "not a number".

`
(from x in _factory.WorksheetRangeNoHeader("A2", "D9", "Invalid Cast") 
where x[2].Cast<double>() > 30.0
select x[2]
`

Actually I couldn't think in another example to throw an exception to test when reading a NoHeader sheet.

Reply:
I took a closer look, and this test doesn't test the same functionality as the first test. In fact, I don't think you *can* test the same functionality. Using the NoHeader methods doesn't do any casting at all. 

Anyway, in your test, `Cast<>()` never actually gets called (which is why it doesn't throw an exception). So, not sure what's going on there :/

Reply:
I really thought about a way to throw an exception while reading a noheader sheet, but I out of ideas.

Reply:
It can be done using mocks (like a mock of `IDataReader` that throws an exception when reading `data[i]`), but that would require quite a bit of refactoring in the project. This is something I'll have to review further.